### PR TITLE
Add django-compressor suport

### PR DESCRIPTION
**Pull request by andresgz**
This PR is intended to solve issue #342 to add support to django-compressor.
It includes a new option to choose it (disabled by default).
According to the selection it will inject the necessary configuration in settings files and base template.

This PR has not been fully tested for all the cases, so if you find any issue, please let me know.

### CODE REVIEW

File: {{cookiecutter.project_slug}}/config/settings/common.py [hunk 1]
```diff
@@ -185,6 +185,8 @@
     'django.contrib.staticfiles.finders.AppDirectoriesFinder',
 )
 
+# 'compressor.finders.CompressorFinder',
+
```
Comment (luzfcb) [inline {{cookiecutter.project_slug}}/config/settings/common.py]: put inside the `{% if cookiecutter.use_compressor == 'y'-%}`

Comment (andresgz) [inline {{cookiecutter.project_slug}}/config/settings/common.py]: thanks @luzfcb,
That was a redundant line, just removed it

**Comment by luzfcb** [comment]
since we use Amazon S3 to staticfiles, I think this is also necessary: 
https://django-compressor.readthedocs.io/en/latest/remote-storages/#django-storages

probably it would also be useful to expose (via environment variable) some of these settings:

https://django-compressor.readthedocs.io/en/latest/settings/#base-settings

since I haven't used it so far, I do not know which of these settings would be good

if any new environment variable is added, It is also necessary to include an example of use in: https://github.com/pydanny/cookiecutter-django/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/env.example


another question: django-compressor has no non-python dependency?
If have any non-python dependencies, it is also necessary to include them in [requirements.apt](https://github.com/pydanny/cookiecutter-django/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/utility/requirements.apt) and [requirements.apt.xenial](https://github.com/pydanny/cookiecutter-django/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/utility/requirements.apt.xenial)

**Comment by andresgz** [comment]
Thanks @luzfcb 
I will add those options for S3 and will include the environment variables.
According to what i read in the documentation, it does not have system requirements.

**Comment by pydanny** [comment]
I believe the Amazon S3 stuff is already defined in https://github.com/pydanny/cookiecutter-django/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/config/settings/production.py#L139-L152. 

Therefore, all that needs to happen is to test that those values (or others in production.py) fulfill the needs of this pull request.

**Comment by chrisdev** [comment]
@andresgz @pydanny in ``production.py`` we probably need to enable  offline compression see https://django-compressor.readthedocs.io/en/latest/scenarios/#offline-compression. 
Especially so if assets are in a CDN. 

We specify need to  ``COMPRESS_OFFLINE = True```

``COMPRESS_STORAGE = 'storages.backends.s3boto.S3BotoStorage'```

In this case we got to run ``manage.py compress``

**Comment by chrisdev** [comment]
@andresgz @pydanny compressor can also handle the SASS work flow. No 
Something like this should be added to the base template.

```<link rel="stylesheet" type="text/x-scss" href="{% static 'css/app.scss' %}">```


and something like this  to ```production.py```

```
COMPRESS_CSS_FILTERS = [
    'compressor.filters.css_default.CssAbsoluteFilter',
    'compressor.filters.cssmin.CSSMinFilter',
]
```

**Comment by chrisdev** [comment]
@andresgz @pydanny here are couple of additional dependencies that will leverage the ```django-compressor``` and get rid of Grunt from the point of view the SASS work flow https://github.com/torchbox/django-libsass and of course libsass

**Comment by andresgz** [comment]
Thanks @chrisdev !
As a personal opinion I would like to leave the compressor configuration as simple possible. And after project is cooked, the user can extend the capabilities with other options.

**Comment by pydanny** [comment]
@andresgz, simplest can be good, but we need an approach tied to tools already in this project, specifically SASS and other options @chrisdev mentions. 

In other words, we want/need an opinionated approach.

**Comment by andresgz** [comment]
Hi guys! 
As a personal approach I prefer to use Gulp tasks for front-end development and compression of SASS and JS assets, to also to take advantage of tools like [Browsersync](https://www.browsersync.io/). Then I use compressor at the end, to compress the final assets. But I think It would be great if @chrisdev can help us to include those options in another PR, that could be very useful when not using these tools in the front side. What do you think?

**Comment by andresgz** [comment]
BTW, just added the missing options in a last commit and also merged the last changes from master. It seems working fine

**Comment by andresgz** [comment]
Hi guys,
I think this is ready to merge. Do you want me to add something extra?

**Comment by pydanny** [comment]
@andresgz I'll be testing your pull request today both locally and in production and will give feedback and we'll hopefully get this merged in ASAP. 😄

**Comment by pydanny** [comment]
I'm going to merge this in so users can try it. Considering that this touches static media and involves configuration, there are going to be a few issues, but nothing that shouldn't be easy to walk people through.

Thanks so much @andresgz!

**Comment by andresgz** [comment]
Thanks @pydanny 
I will be pending in case there is any issue.

**Pull request merged**